metricflow-semantics/metricflow_semantics/query/validation_rules/metric_time_requirements.py

```diff
@@ -46,77 +66,95 @@ def __init__(  # noqa: D107
             )
         )
 
-    @cached_property
-    def _group_by_items_include_metric_time(self) -> bool:
-        for group_by_item_input in self._resolver_input_for_query.group_by_item_inputs:
-            if group_by_item_input.spec_pattern.matches_any(self._metric_time_specs):
-                return True
-
-        return False
-
-    def _query_includes_metric_time_or_agg_time_dimension(self, metric_reference: MetricReference) -> bool:
-        return self._group_by_items_include_metric_time or self._group_by_items_include_agg_time_dimension(
-            query_resolver_input=self._resolver_input_for_query, metric_reference=metric_reference
-        )
-
-    def _group_by_items_include_agg_time_dimension(
+    @lru_cache
```

Comment:
Just checking - I think Paul said it was fine to use this `lru_cache` becasue the number of items is small enough?